File: module-main/src/main/java/com/huyingbao/module/main/ui/main/MainFragment.java
package com.huyingbao.module.main.ui.main;

import android.os.Bundle;

import com.huyingbao.core.custom.CommonFragment;
import com.huyingbao.core.scope.PerActivity;
import com.huyingbao.module.main.R;
import com.huyingbao.module.main.R2;
import com.huyingbao.module.main.action.MainActionCreator;
import com.huyingbao.module.main.action.MainActions;

import javax.inject.Inject;

import butterknife.OnClick;

/**
 * Created by liujunfeng on 2017/12/7.
 */
@PerActivity
public class MainFragment extends CommonFragment {
    @Inject
    MainActionCreator mActionCreator;

    @Inject
    public MainFragment() {
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_main;
    }

    @Override
    public void afterCreate(Bundle savedInstanceState) {
        initActionBar(null, false);
    }

    @OnClick(R2.id.btn_main_to_product)
    public void toProductList() {
        mActionCreator.postLocalAction(MainActions.TO_PRODUCT_LIST);
    }

    @OnClick(R2.id.btn_main_to_github)
    public void toGithub() {
        mActionCreator.postLocalAction(MainActions.TO_GITHUB);
    }
}

File: module-github/src/main/java/com/huyingbao/module/github/ui/main/view/MainFragment.kt
package com.huyingbao.module.github.ui.main.view

import android.os.Bundle
import com.huyingbao.core.base.fragment.BaseRxFragment
import com.huyingbao.module.github.R
import com.huyingbao.module.github.ui.main.store.MainStore

class MainFragment : BaseRxFragment<MainStore>() {
    companion object {
        fun newInstance(): MainFragment {
            return MainFragment()
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.github_fragment_main
    }

    override fun afterCreate(savedInstanceState: Bundle?) {
    }
}